packages/core/strapi/lib/types/factories.d.ts

```diff
@@ -62,42 +53,41 @@ type ControllerCallback<T extends GenericController = GenericController> = (para
 }) => T;
 type ServiceCallback<T extends GenericService = GenericService> = (params: { strapi: Strapi }) => T;
 
-export declare function createCoreRouter<T extends SchemaUID>(
+export declare function createCoreRouter<T extends UID.ContentType>(
```

Comment:
Since we have multiple UID namespaces (one that's based on the shared registries, and the other that represents generic UIDs), we should make sure we're using the namespaces they're part of too.
Also, here I think you might want to use Common.UID (<-- contains union type of content-types) instead of the core UID namespace(<-- doesn't have auto-completion since it's only a union type of template strings)